tests/base.py

```diff
@@ -429,5 +410,36 @@ def _load_schemas(self, stream, shared: bool = False):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.start_date = self.get_properties().get('start_date')
-        self.maxDiff=None
+        self.start_date = self.get_properties().get("start_date")
+        self.maxDiff = None
+
+    def create_interrupt_sync_state(
+        self, state: Dict, interrupt_stream: str, pending_streams: Set, start_date: str
+    ) -> Dict:
+        """This function will create a new interrupt sync bookmark state."""
```

Comment:
Be consistent with argument typing.

Reply:
i have added typehints to majority of the functions